storagenode/piecestore/verification.go

```diff
@@ -26,41 +28,42 @@ var (
 
 // VerifyOrderLimit verifies that the order limit is properly signed and has sane values.
 // It also verifies that the serial number has not been used.
-func (endpoint *Endpoint) VerifyOrderLimit(ctx context.Context, limit *pb.OrderLimit) (err error) {
+func (endpoint *Endpoint) verifyOrderLimit(ctx context.Context, limit *pb.OrderLimit) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
 	// sanity checks
 	now := time.Now()
 	switch {
 	case limit.Limit < 0:
-		return ErrProtocol.New("order limit is negative")
+		return status.Error(codes.InvalidArgument, "order limit is negative")
 	case endpoint.signer.ID() != limit.StorageNodeId:
-		return ErrProtocol.New("order intended for other storagenode: %v", limit.StorageNodeId)
+		return status.Errorf(codes.InvalidArgument, "order intended for other storagenode: %v", limit.StorageNodeId)
 	case endpoint.IsExpired(limit.PieceExpiration):
-		return ErrProtocol.New("piece expired: %v", limit.PieceExpiration)
+		return status.Errorf(codes.InvalidArgument, "piece expired: %v", limit.PieceExpiration)
 	case endpoint.IsExpired(limit.OrderExpiration):
-		return ErrProtocol.New("order expired: %v", limit.OrderExpiration)
+		return status.Errorf(codes.InvalidArgument, "order expired: %v", limit.OrderExpiration)
 	case now.Sub(limit.OrderCreation) > endpoint.config.OrderLimitGracePeriod:
-		return ErrProtocol.New("order created too long ago: %v", limit.OrderCreation)
+		return status.Errorf(codes.InvalidArgument, "order created too long ago: %v", limit.OrderCreation)
 	case limit.SatelliteId.IsZero():
-		return ErrProtocol.New("missing satellite id")
+		return status.Errorf(codes.InvalidArgument, "missing satellite id")
 	case limit.UplinkPublicKey.IsZero():
-		return ErrProtocol.New("missing uplink public key")
+		return status.Errorf(codes.InvalidArgument, "missing uplink public key")
 	case len(limit.SatelliteSignature) == 0:
-		return ErrProtocol.New("missing satellite signature")
+		return status.Errorf(codes.InvalidArgument, "missing satellite signature")
 	case limit.PieceId.IsZero():
-		return ErrProtocol.New("missing piece id")
+		return status.Errorf(codes.InvalidArgument, "missing piece id")
 	}
 
 	if err := endpoint.trust.VerifySatelliteID(ctx, limit.SatelliteId); err != nil {
-		return ErrVerifyUntrusted.Wrap(err)
+		return status.Errorf(codes.FailedPrecondition, "untrusted: %+v", err)
```

Comment:
Wondering whether `codes.PermissionDenied` would be more appropriate here.

Reply:
I thought about it but I believe that this was a better fit.

Reply:
This is not a precondition failure, it's a permission issue. The specific satellite is not allowed to do any operations on this server, hence the PermissionDenied.

Reply:
I saw it the for dealing with a satellite it has to be in the whitelist that's the reason for the failed precondition, considering that the connection comes from the uplink.

But, yes, seeing from that point of view the `PermissionDenied` looks better.